tests/functional/h/views/admin/test_permissions.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+
+
+class TestAdminPermissions:
+    PAGES = (
+        # URL, accessible by role.Staff?
+        ("/admin/", True),
+        ("/admin/admins", False),
+        ("/admin/badge", False),
+        ("/admin/features", False),
+        ("/admin/groups", True),
+        ("/admin/mailer", True),
+        ("/admin/nipsa", False),
+        ("/admin/oauthclients", False),
+        ("/admin/organizations", True),
+        ("/admin/staff", False),
+        ("/admin/users", True),
+        ("/admin/search", False),
+    )
+
+    @pytest.mark.parametrize("url", (page[0] for page in PAGES))
+    def test_not_accessible_by_regular_user(self, app, url, user):
+        self.login(app, user)
+        
+        res = app.get(url, expect_errors=True)
+
+        assert res.status_code == 404
+
+    @pytest.mark.parametrize("url", (page[0] for page in PAGES))
+    def test_accessible_by_admin(self, app, url, admin_user):
+        self.login(app, admin_user)
+
+        res = app.get(url)
+
+        assert res.status_code == 200
+
+    @pytest.mark.parametrize("url,accessible", PAGES)
+    def test_accessible_by_staff(self, app, url, accessible, staff_user):
+        self.login(app, staff_user)
+
+        res = app.get(url, expect_errors=not accessible)
+
+        assert res.status_code == 200 if accessible else 404
+
+    @pytest.fixture
+    def user(self, db_session, factories):
+        # Password is 'pass'
+        return factories.User(
+            password="$2b$12$21I1LjTlGJmLXzTDrQA8gusckjHEMepTmLY5WN3Kx8hSaqEEKj9V6"
```

Comment:
The linkage between this string and the "pass" below is extremely obscure. Is there a convenient way to generate this string from "pass"? We could import the `UserPasswordService` which actually takes no arguments. Not sure if that is verboten though.